tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -105,7 +107,61 @@ propagationLevel is not (DbmPropagationLevel.Service or DbmPropagationLevel.Full
             // modify the command to add the comment
             var commandText = command.CommandText ?? string.Empty;
             var propagationComment = StringBuilderCache.GetStringAndRelease(propagatorStringBuilder);
-            if (ShouldAppend(integrationId, commandText))
+            if (command.CommandType == CommandType.StoredProcedure && integrationId == IntegrationId.SqlClient) // only supported for Microsoft SQL Server
+            {
+                // Save the original stored procedure name
+                string procName = command.CommandText ?? string.Empty;
+
+                if (string.IsNullOrEmpty(procName))
+                {
+                    return false;
+                }
+
+                // Build parameter list for EXEC statement
+                StringBuilder paramList = new StringBuilder();
+                foreach (DbParameter? param in command.Parameters)
+                {
+                    if (param == null)
+                    {
+                        continue;
+                    }
+
+                    // Skip return value parameters
+                    if (param.Direction == ParameterDirection.ReturnValue)
+                    {
+                        continue;
```

Comment:
TODO comment me
https://learn.microsoft.com/en-us/sql/relational-databases/stored-procedures/return-data-from-a-stored-procedure?view=sql-server-ver16
A `RETURN` parameter isn't needed in the `EXEC` it's handled automatically

Note that we don't actually change the parameters, all of the parameters remain on the command, we just changing the type to `TEXT` and building up the `EXEC` command.